src/Bicep.Core/Registry/Oci/OciRegistryContentClient.cs

```diff
@@ -0,0 +1,110 @@
+        _endpoint = endpoint;
+        _repositoryName = repositoryName;
+    }
+
+    public OciRegistryContentClient(Uri endpoint, string repositoryName, TokenCredential credential, ContainerRegistryClientOptions options)
+    {
+        _client = new ContainerRegistryContentClient(endpoint, repositoryName, credential, options);
+        _endpoint = endpoint;
+        _repositoryName = repositoryName;
+    }
+
+    #region ContainerRegistryContentClient functionality forwarded
+
+    public Task<Response<SetManifestResult>> SetManifestAsync(OciImageManifest manifest, string? tag = default, ManifestMediaType? mediaType = default, CancellationToken cancellationToken = default)
+        => _client.SetManifestAsync(manifest, tag, mediaType, cancellationToken);
+
+    public Task<Response<SetManifestResult>> SetManifestAsync(BinaryData manifest, string? tag = default, ManifestMediaType? mediaType = default, CancellationToken cancellationToken = default)
+        => _client.SetManifestAsync(manifest, tag, mediaType, cancellationToken);
+
+    public Task<Response<UploadRegistryBlobResult>> UploadBlobAsync(Stream content, CancellationToken cancellationToken = default)
+        => _client.UploadBlobAsync(content, cancellationToken);
+
+    public Task<Response<DownloadRegistryBlobResult>> DownloadBlobContentAsync(string digest, CancellationToken cancellationToken = default)
+        => _client.DownloadBlobContentAsync(digest, cancellationToken);
+
+    public Task<Response<GetManifestResult>> GetManifestAsync(string tagOrDigest, CancellationToken cancellationToken = default)
+        => _client.GetManifestAsync(tagOrDigest, cancellationToken);
+
+    #endregion ContainerRegistryContentClient functionality forwarded
+
+    public async Task<IEnumerable<(string digest, string? artifactType)>> GetReferrersAsync(string mainManifestDigest)
+    {
+        Debug.Assert(mainManifestDigest != null);
+
+        IEnumerable<(string digest, string? artifactType)>? referrers = null;
+
+        var request = _client.Pipeline.CreateRequest();
+        request.Method = RequestMethod.Get;
+        request.Uri.Reset(_endpoint);
+        request.Uri.AppendPath("/v2/", false);
+        request.Uri.AppendPath(_repositoryName, true);
+        request.Uri.AppendPath("/referrers/", false);
+        request.Uri.AppendPath(mainManifestDigest);
+
+        var response = await _client.Pipeline.SendRequestAsync(request, CancellationToken.None);
+        if (response.IsError)
+        {
+            throw new Exception($"Unable to retrieve source manifests. Referrers API failed with status code {response.Status}");
+        }
+
+#pragma warning disable IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
+        var referrersResponse = JsonSerializer.Deserialize<JsonElement>(response.Content.ToString());
```

Comment:
To avoid the IL2026 warning:
```suggestion
        var referrersResponse = JsonElementFactory.Create(response.Content.ToStream());
```